app/services/api/declarations/clawback.rb

```diff
@@ -0,0 +1,62 @@
+module API::Declarations
+  class Clawback
+    include API::Concerns::Declarations::SharedAction
+
+    validate :paid
+    validate :not_already_refunded
+    validate :output_fee_statement_available
+
+    def clawback
+      return false unless valid?
+
+      Declarations::Clawback.new(
+        author:,
+        declaration:,
+        next_available_output_fee_statement:
+      ).clawback
+    end
+
+  private
+
+    def paid
+      return if errors[:declaration_id].any?
+      return if declaration.paid?
+
+      errors.add(:declaration_id, "The declaration must be paid before it can be clawed back")
+    end
+
+    def not_already_refunded
+      return if errors[:declaration_id].any?
+      return if declaration.clawback_status_no_clawback?
+
+      errors.add(:declaration_id, "The declaration will or has been refunded")
+    end
+
+    def output_fee_statement_available
+      return if errors[:declaration_id].any?
+      return if next_available_output_fee_statement.present?
+
+      no_output_fee_statement_error_message = <<~TXT.squish
+        You cannot submit or void declarations for the #{contract_period_year}
+        cohort. The funding contract for this cohort has ended. Get in touch
+        if you need to discuss this with us
+      TXT
+      errors.add(:declaration_id, no_output_fee_statement_error_message)
```

Comment:
I think we may need to remap `declaration_id` to `participant_declaration` to match ECF as well